runtime/near-vm-logic/src/logic.rs

```diff
@@ -1147,26 +1154,35 @@ impl<'a> VMLogic<'a> {
         pub_key_len: u64,
         pub_key_ptr: u64,
     ) -> Result<u64> {
-        use ed25519_dalek::{PublicKey, Signature, Verifier, SIGNATURE_LENGTH};
+        use ed25519_dalek::{PublicKey, Signature, Verifier, PUBLIC_KEY_LENGTH, SIGNATURE_LENGTH};
 
         self.gas_counter.pay_base(ed25519_verify_base)?;
+        let msg = self.get_vec_from_memory_or_register(msg_ptr, msg_len)?;
+        let signature_array = self.get_vec_from_memory_or_register(sig_ptr, sig_len)?;
         if sig_len != SIGNATURE_LENGTH as u64 {
```

Comment:
I think this is incorerct, this should be checking `signature_array.len()` rather than `sig_len`. `sig_len != signature_array.len()` if the data is loaded from a register. 

Could you add tests for signatures and public keys of incorrect lenght being loaded from a register?

Reply:
> Could you add tests for signatures and public keys of incorrect lenght being loaded from a register?
How do I "force" the data to be loaded from a register? @matklad

Reply:
Set `len` to `u64::MAX` and `ptr` to  the index of the registed. See also `read_register` and `write_register` host functions.

Reply:
I'm trying

```rs
    let result = logic.ed25519_verify(
        u64::MAX,
        signature.as_ptr() as _,
        message.len() as _,
        message.as_ptr() as _,
        public_key.len() as _,
        public_key.as_ptr() as _,
    );
```
But I get `Err(HostError(InvalidRegisterId { register_id: 6107169072 }))`
How do I get the right register?

Reply:
You need to set ptr to the index of the registed. So, somethin like `logic.ed25519_verify(u64::MAX, 1, ..)`, where previously you called `write_register` to file the `1` register with signature's bytes

Reply:
thanks